cogs/cryochamber.py

```diff
@@ -0,0 +1,108 @@
+"""Commands for preserving Discord messages in another channel."""
+import re
+
+from discord import Embed, Colour
+from discord.ext.commands import (Cog, CommandError, MessageConverter,
+                                  TextChannelConverter, group, check)
+
+
+class Cryochamber(Cog):
+    def __init__(self, _):
+        self.ongoing_preservations = set()
+
+    async def can_manage_messages(ctx):
+        get_channel = TextChannelConverter().convert
+        arg = ctx.message.content
+        exp = r'(?P<source_specifier>.*?) in (?P<destination_channel_specifier>.*?)'
+        command_match = re.fullmatch(exp, arg)
+        dest_channel = await get_channel(ctx,
+                                         command_match.group('destination_channel_specifier'))
```

Comment:
command_match might be `None` if there was no match, which causes an exception here.

Reply:
This might require refactoring the manage messages check, since we are doing a granular check per channel rather than a server check. I probably shouldn't have suggested making this a check at all :(